includes/Modules/Analytics_4/Reset_Audiences.php

```diff
@@ -155,8 +155,11 @@ public function reset_audience_data() {
 					}
 				}
 
-				// Reset the users Audience Settings, such as configured audiences.
-				$this->audience_settings->reset();
+				// Reset the users audience settings, such as configured audiences.
+				$audience_settings_keys = array_keys( $this->audience_settings->get() );
+				foreach ( $audience_settings_keys as $key ) {
+					$this->audience_settings->delete( $key );
+				}
```

Comment:
According to what I can see, the `Audience_Settings` class or its parent `User_Setting` does not have a `delete` method that accepts any parameters.

Instead, `User_Setting::delete` is a simple method that entirely deletes the `googlesitekit_audience_settings` `user_meta` option for the current user, which should be sufficient in our case. What do you think?

```suggestion
				$this->audience_settings->delete();
```